link/kprobe_test.go

```diff
@@ -283,3 +286,101 @@ func TestDetermineRetprobeBit(t *testing.T) {
 	c.Assert(err, qt.IsNil)
 	c.Assert(rpu, qt.Equals, uint64(0))
 }
+
+func TestKprobeProgramCall(t *testing.T) {
+	var (
+		// Map key
+		key uint32
+		// Map value
+		val uint32
+	)
+
+	// Create ebpf map. Will contain only one key with initial value 0.
+	m, err := ebpf.NewMap(&ebpf.MapSpec{
+		Name:       "kprobetestmap",
+		Type:       ebpf.Array,
+		KeySize:    4,
+		ValueSize:  4,
+		MaxEntries: 1,
+		Contents: []ebpf.MapKV{
+			{Key: key, Value: val},
+		},
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Create ebpf program. When called, will set the value of key 0 in
+	// the map created above to 1.
+	p, err := ebpf.NewProgram(&ebpf.ProgramSpec{
+		Name: "kprobetestprog",
+		Type: ebpf.Kprobe,
+		Instructions: asm.Instructions{
+			// u32 key = 0
+			asm.Mov.Imm(asm.R1, 0),
+			asm.StoreMem(asm.RFP, -4, asm.R1, asm.Word),
+
+			// u32 val = 1
+			asm.Mov.Imm(asm.R1, 1),
+			asm.StoreMem(asm.RFP, -8, asm.R1, asm.Word),
+
+			// bpf_map_update_elem(...)
+			asm.Mov.Reg(asm.R2, asm.RFP),
+			asm.Add.Imm(asm.R2, -4),
+			asm.Mov.Reg(asm.R3, asm.RFP),
+			asm.Add.Imm(asm.R3, -8),
+			asm.LoadMapPtr(asm.R1, m.FD()),
+			asm.Mov.Imm(asm.R4, 0),
+			asm.FnMapUpdateElem.Call(),
+
+			// exit 0
+			asm.Mov.Imm(asm.R0, 0),
+			asm.Return(),
+		},
+		License:   "Dual MIT/GPL",
+		ByteOrder: binary.LittleEndian,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Assert that the value is 0 as set earlier.
+	if err := m.Lookup(key, &val); err != nil {
```

Comment:
Yes, usually I put "obvious" assertions like this one to make clear what the test is trying to accomplish but I think I can remove this since it's written in the comments